tests/api_fastapi/test_app.py

```diff
@@ -19,6 +19,15 @@
 from unittest import mock
 
 
+def test_main_app_lifespan(client):
+    with client() as test_client:
+        test_app = test_client.app
+
+        # assert the app was created and lifespan was called
+        assert test_app
+        assert test_app.state.lifespan_called, "Lifespan not called on Execution API app."
```

Comment:
Does each app have it's own state object? If so I don't think we are asserting the subapp's lifespan is called here.

Reply:
No indeed :) That's why the test is called `test_main_app_lifespan` and there's one for the execution app below.